packages/data-context/src/sources/FileDataSource.ts

```diff
@@ -36,7 +36,11 @@ export class FileDataSource {
       // The working directory path may include characters that conflict with glob
       // syntax (brackets, parentheses, etc.) and cause our searches to inadvertently fail.
       // We scope our search to the working directory using the `cwd` globby option.
-      return globPattern.replace(cwd, '.')
+      if (globPattern.startsWith(cwd)) {
+        return globPattern.replace(cwd, '.')
+      }
```

Comment:
A root directory of `/cypress/e2e/`, would still break right? `/cypress/e2e/cypress/e2e/*.cy.js` -> `*.cy.js`

Reply:
Is there any way you can just replace the first instance of the cwd?

Reply:
Using `replace` (vs. `replaceAll`), only the first instance will be removed: `/cypress/e2e/cypress/e2e/*.cy.js` -> `./cypress/e2e/*.cy.js`

Reply:
So how did we see the issue initially? does this get called multiple times?

Reply:
So in the issue, there's a working directory of `/e2e`, and a glob pattern of `cypress/support/e2e.{js,ts}`. Without the startsWith check, we get `cypress/support..{js.ts}`, which is not helpful.